external-crates/move/crates/move-compiler/src/expansion/aliases.rs

```diff
@@ -43,27 +43,51 @@ impl AliasSet {
     }
 }
 
-macro_rules! resolve_alias {
-    ($map:expr, .$namespace:ident, $name:expr) => {{
-        let mut current_scope = Some($map);
-        let name = $name;
-        loop {
-            let Some(scope) = current_scope else {
-                break None;
-            };
-            let Some(entry) = scope.$namespace.get(name).copied() else {
-                current_scope = scope.previous.as_mut().map(|x| &mut **x);
-                continue;
-            };
-            // Note, might have already been removed by a different `$namespace` resolution
-            scope.unused.remove(&(*name, entry).into());
-
-            let original_name = scope.$namespace.get_full_key(&name).unwrap();
-            break Some((original_name, entry));
+macro_rules! decl_resolve_alias {
+    ($f:ident, $fimpl: ident, .$namespace:ident, $t:ty) => {
+        impl AliasMap {
+            fn $fimpl(
+                &mut self,
+                name: &Name,
+                mut k: impl FnMut(&mut Self, &Name, &$t),
+            ) -> Option<(Name, $t)> {
+                let mut current_scope = Some(self);
+                loop {
+                    let Some(scope) = current_scope else {
+                        break None;
+                    };
+                    let Some(entry) = scope.$namespace.get(name).copied() else {
+                        current_scope = scope.previous.as_mut().map(|x| &mut **x);
+                        continue;
+                    };
+                    let original_name = scope.$namespace.get_full_key(&name).unwrap();
+                    k(scope, &original_name, &entry);
```

Comment:
`k` for continuation?

Reply:
Something like that. I just had already used `f` for the function name